shell/packages/sandstorm-permissions/permissions.js

```diff
@@ -15,6 +15,18 @@
 // limitations under the License.
 
 var Crypto = Npm.require("crypto");
+var Url = Npm.require("url");
+
+SandstormPermissions = function (db) {
+  function cleanupSelfDestructing() {
+    var now = new Date();
+    db.collections.apiTokens.remove({expiresIfUnused: {$lt: now}});
+  };
+
+  // Make self-destructing tokens actually self-destruct, so they don't
+  // clutter the token list view.
+  SandstormDb.periodicCleanup(5 * 60 * 1000, cleanupSelfDestructing);
```

Comment:
I guess what I don't like here is largely aesthetic, but basically I think you should only have an object if it encapsulates some meaningful state. So, yeah, I think the right model is indeed to have a function called "SandstormPermissions.registerCleanups(db)", which is called from db-deprecated.js (which perhaps we should consider renaming to "init-server.js" or something at some point, though not yet).

I guess the alternative is to say that SandstormPermissions represents a "component" of the system. It's only intended that you have one instance of it per database, and all its functionality is exposed as methods of that instance rather than "static" functions. This allows replacing it with a mock instance and could also allow adding some state someday, like perhaps a "configuration" that somehow affects the logic. I guess this is what you were going for originally, from what you said in meetings.

So I guess I'd say, if all the methods are "static", then there shouldn't be a constructor. But if the methods are methods, then it makes sense to have a constructor that registers the cleanup. (Ideally there'd also be some sort of "destructor" that unregisters cleanups... but... >_>)